src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/TestExtensions.cs

```diff
@@ -23,45 +23,96 @@ public class TestExtensions
         internal Dictionary<string, TestDiscovererPluginInformation> TestDiscoverers { get; set; }
 
         /// <summary>
+        /// Gets or sets a value indicating whether are test discoverers cached.
+        /// </summary>
+        internal bool AreTestDiscoverersCached { get; set; }
+
+        /// <summary>
         /// Gets or sets test executor extensions.
         /// </summary>
         internal Dictionary<string, TestExecutorPluginInformation> TestExecutors { get; set; }
 
         /// <summary>
+        /// Gets or sets a value indicating whether are test executors cached.
+        /// </summary>
+        internal bool AreTestExecutorsCached { get; set; }
+
+        /// <summary>
         /// Gets or sets test setting provider extensions.
         /// </summary>
         internal Dictionary<string, TestSettingsProviderPluginInformation> TestSettingsProviders { get; set; }
 
         /// <summary>
+        /// Gets or sets a value indicating whether are test settings providers cached.
+        /// </summary>
+        internal bool AreTestSettingsProvidersCached { get; set; }
```

Comment:
Are we planning to move these out to their respective plugin managers? This is becoming a God class.